libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/report/model/aeaa/AeaaMatchableDetailsAmbDataClass.java

```diff
@@ -30,529 +27,603 @@
-                .filter(e -> e.getKey() instanceof AeaaAdvisoryTypeIdentifier)
-                .collect(Collectors.toMap(e -> (AeaaAdvisoryTypeIdentifier<?>) e.getKey(), Map.Entry::getValue)));
-        this.addOtherReferencedIds(legacyReferencedIds.entrySet().stream()
-                .filter(e -> e.getKey() instanceof AeaaOtherTypeIdentifier)
-                .collect(Collectors.toMap(e -> (AeaaOtherTypeIdentifier) e.getKey(), Map.Entry::getValue)));
-    }
+public abstract class AeaaMatchableDetailsAmbDataClass<AMB extends AbstractModelBase,
+		DC extends AeaaMatchableDetailsAmbDataClass<AMB, DC>> extends
+		AeaaAmbDataClass<AMB, DC> {
+
+	protected final static Set<String> CONVERSION_KEYS_AMB =
+			new HashSet<String>(AeaaAmbDataClass.CONVERSION_KEYS_AMB) {{
+				add(AdvisoryMetaData.Attribute.MATCHING_SOURCE.getKey());
+				add(AdvisoryMetaData.Attribute.DATA_SOURCE.getKey());
+
+				add(AdvisoryMetaData.Attribute.REFERENCED_IDS.getKey());
+				add(AeaaInventoryAttribute.VULNERABILITY_REFERENCED_CONTENT_IDS.getKey());
+
+				add(AdvisoryMetaData.Attribute.REFERENCED_SECURITY_ADVISORIES.getKey());
+				add(VulnerabilityMetaData.Attribute.REFERENCED_SECURITY_ADVISORIES.getKey());
+				add(AdvisoryMetaData.Attribute.REFERENCED_VULNERABILITIES.getKey());
+				add(VulnerabilityMetaData.Attribute.REFERENCED_VULNERABILITIES.getKey());
+				add(AdvisoryMetaData.Attribute.REFERENCED_OTHER.getKey());
+				add(VulnerabilityMetaData.Attribute.REFERENCED_OTHER.getKey());
+			}};
+	protected final static Set<String> CONVERSION_KEYS_MAP =
+			new HashSet<String>(AeaaAmbDataClass.CONVERSION_KEYS_MAP) {{
+				add("source");
+				add("sourceImplementation");
+				add("dataFillingSources");
+				add("matchingSources");
+				add("referencedIds");
+				add("referencedSecurityAdvisories");
+				add("referencedVulnerabilities");
+				add("referencedOtherIds");
+			}};
+	private static final Logger LOG = LoggerFactory.getLogger(AeaaMatchableDetailsAmbDataClass.class);
+	protected final Map<AeaaVulnerabilityTypeIdentifier<?>, Set<String>> referencedVulnerabilities = new HashMap<>();
+	protected final Map<AeaaAdvisoryTypeIdentifier<?>, Set<String>> referencedSecurityAdvisories = new HashMap<>();
+	protected final Map<AeaaOtherTypeIdentifier, Set<String>> referencedOtherIds = new HashMap<>();
+
+	protected final List<AeaaDataSourceIndicator> matchingSources = new ArrayList<>();
+	protected final Set<AeaaContentIdentifierStore.AeaaContentIdentifier> dataSources = new HashSet<>();
+
+	/**
+	 * Artifacts that are affected by this entry.<br> Will not be written back into the inventory as a field, but
+	 * rather
+	 * as references from the artifact to this entry.
+	 * <pre>
+	 * Key:   Artifact field name
+	 * Value: Set&lt;Artifact&gt; affected artifacts
+	 * </pre>
+	 */
+	protected final Map<String, Set<Artifact>> affectedArtifacts = new HashMap<>();
+
+	protected final Set<AssetMetaData> affectedAssets = new HashSet<>();
```

Comment:
This information should not be stored in a property, but rather recalculated upon any access to the vulnerability instance using the `affectedArtifacts`.
Storing this information would surely lead to sync-issues in the future, having to always update the information in two properties (redundancy leads to inconsistency).

Also, if we were to update this file, the according class in the artifact analysis project would have to be updated as well.